transport/redis/outbound.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package redis
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/redis/internal"
+)
+
+type outbound struct {
+	client QueueClient
+	deps   transport.Deps
+
+	startOnce sync.Once
+	stopOnce  sync.Once
```

Comment:
nit: If you don't want to name these `startOnce` and `stopOnce`, you can use
an anonymous struct here.

```
type outbound struct {
    // ...
    
    once struct {
        Start sync.Once
        Stop sync.Once
    }
}
```

Usage is then,

```
func (o *outbound) Start(...) error {
    o.once.Start.Do(func() {
        // ..
    })
}
```